### PR TITLE
Service portal/documents modal

### PR DESCRIPTION
**Pull request by andriandresson**
# \<Document exception modal\>

Attach a link to issue if relevant
https://app.asana.com/0/1199218948638353/1199195379824200

## What

- Displaying exception modal to notify the user that external documents are not supported.
- Updated Modal- & ModalBase components.

## Why
- External documents have broken URL's.
- In order to have correct state of the modal visibility, I had to add optional cb func that is fired when the modal visibility is hidden, either by an CTA button or when the user clicks on the DialogBackdrop.

#### Desktop: 
image

#### Mobile: 
image

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by Thorrik58 — APPROVED** [review]

File: libs/island-ui/core/src/lib/ModalBase/ModalBase.tsx [hunk 1]
```diff
@@ -71,6 +76,10 @@ export const ModalBase: FC<ModalBaseProps> = ({
     if (toggleClose) closeModal()
   }, [toggleClose])
 
+  useEffect(() => {
+    if (!modal.visible && onClose) onClose()
```
Comment (barabrian) [inline libs/island-ui/core/src/lib/ModalBase/ModalBase.tsx:80]: Would it make sense to have this something like?
```suggestion
    onChange(modal.visible)
```
or maybe `onVisibilityChange`, just so we won't need to extend this in the future and change the API, or will that not work for your use case?

**Review by barabrian — COMMENTED** [review]
LGTM, minor comments

File: libs/service-portal/documents/src/components/DocumentCard/DocumentCard.treat.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { style } from 'treat'
+import { themeUtils } from '@island.is/island-ui/theme'
+
+export const modalButtonWrapper = style({
+  ...themeUtils.responsiveStyle({
+    sm: {
+      width: '100%',
+    },
+    lg: {
+      width: 'fit-content',
```
[the remark this answers is not in the document]
Comment (andriandresson) [reply libs/service-portal/documents/src/components/DocumentCard/DocumentCard.treat.ts:10]: According to [our handbook](https://github.com/island-is/island.is/blob/main/handbook/technical-overview/technical-direction.md) I believe we do not have to support IE  😅

**Review by barabrian — DISMISSED** [review]
LGTM!

File: libs/island-ui/core/src/lib/ModalBase/ModalBase.tsx (hunk 1 above)
Comment (andriandresson) [reply libs/island-ui/core/src/lib/ModalBase/ModalBase.tsx]: Good point 👍

**Review by Thorrik58 — APPROVED** [review]

**Review by atligudl — APPROVED** [review]

**Pull request merged**